plugin/plugin.go

```diff
@@ -65,6 +79,21 @@ func (p *plugin) Convert(ctx context.Context, req *converter.Request) (*drone.Co
 		return nil, nil
 	}
 
+	convertStart := time.Now()
+	metrics.ConversionCount.Inc()
+	droneCfg, err := p.doConversion(ctx, req)
+	if err != nil {
+		p.recordFail()
```

Comment:
instead of putting recordSuccess and recordFail inside the function, we can actually wrap the function:

```golang
// WithMetrics returns a new converter plugin.
func WithMetrics(base converter.Plugin) converter.Plugin {
	return &withMetrics{base: base}
}

type withMetrics struct {
	base *repo.Registry
}

func (p *withMetrics) Convert(ctx context.Context, req *converter.Request) (*drone.Config, error) {
	start := time.Now()
	res, err := p.base.Convert(ctx, req)
	if err != nil {
		metrics.ConversionFailCount.Inc()
	} else {
		dur := time.Now().Sub(start)
		metrics.ConversionLatency.Observe(dur.Seconds())
		metrics.ConversionSuccessCount.Inc()
	} 
	return res, err
}
```

I see this being useful for most extensions, so it is something I could eventually abstract to our shared library.